### DISCUSSION TITLE
Support for TRNG in ESP32?

### DISCUSSION THREAD

**Discussion by wzab** [micropython/micropython#11728] [ESP32] [answered]
The ESP32 chips offer the True Random Number Generator (TRNG) also known as Hardware Random Number Generator. It is documented on that page: https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/system/random.html 
I think, it would be good to add support for this feature in MicroPython. The true random numbers are important especially for cryptographic protocols.
Do you think it is a good idea? Is anybody working on it?

**Comment by robert-hh** [comment, marked as answer]
The TRNG of the ESP32 and it's interface via esp_random() is already used by the os.urandom() method.

**Reply by wzab** [reply]
Thank you! I didn't know that.
Indeed, the documentation states:

> [os.urandom(n)](https://docs.micropython.org/en/latest/library/os.html#os.urandom)
> Return a bytes object with n random bytes. Whenever possible, it is generated by the hardware random number generator.
